src/main/kotlin/org/rust/lang/core/resolve/Processors.kt

```diff
@@ -622,12 +622,15 @@ fun filterDeriveProcMacros(processor: RsResolveProcessor): RsResolveProcessor =
         true
     }
 
-fun RsOuterAttributeOwner.shouldHideElementInCompletion(contextMod: RsMod): Boolean {
+fun RsOuterAttributeOwner.shouldHideElementInCompletion(context: RsElement, contextMod: RsMod): Boolean {
     val elementContainingMod = containingMod
     if (elementContainingMod == contextMod) return false
 
     // Hide `#[doc(hidden)]` items
-    if (queryAttributes.isDocHidden) return true
+    val isDeriveMacroInImport = context.ancestorStrict<RsUseItem>() != null
+        && this is RsFunction && isCustomDeriveProcMacroDef
+        && containingCrate != contextMod.containingCrate
```

Comment:
Why do we need the crate check? :thinking: 
If it is the same crate, we should show it anyway (or now show, but it shouldn't depend on `#[doc(hidden)]` value), isn't it?

Reply:
I added crate check because don't want to change any behaviour of completing proc macros within same crate (it is currently working incorrectly - should complete function name and not derive macro name - so it should be fixed separately)